[edn] Use MuBi4False instead of 4'h0 for flag0 when enabling entropy src

vogelpi opened the pull request:
Recently, flag0 got converted from a single bit to a MuBi4 field. When seeing invalid MuBi4 values, the CSRNG hardware triggers a recoverable error and stops giving out entropy. As it turns out, in some places the implementation still assumes 4'h0 was interpreted as a legal MuBi4False, which effectively breaks EDN auto mode and all tests using that such as otbn_randomness_test.

This is related to lowRISC/OpenTitan#15165.

mwbranstad reviewed (commented):
Hi Pirmin, thanks for looking at this change in detail. I am assuming that an unwanted alert is firing from CSRNG, but after looking at the RTL, this seems like the RTL is ok. No change should be needed to the flag0 field (bits 11:8 of app cmd) for a GEN command. The flag0 field is only needed to be valid for a INS and RES command.

vogelpi commented:
Thanks for the quick feedback @mwbranstad ! Indeed, the hjson/RTL change doesn't seem necessary as flag0 is used for the instantiate and reseed commands only according to the documentation. I can revert this unless we want to leave it to keep things aligned.

In contrast, the SW change of this PR is required. Otherwise EDN auto mode and all tests using it are broken because of the alert that is triggered inside CSRNG.

tjaychen commented:
> Oh, now I understand why CI fails. The edn_unittest times out...

yeah that's right. You'll need to change quite a number of things in both the csrng and edn unittests. 
I got annoyed...so just left the 0 case in there :)

vogelpi commented:
EDN unittests should now work. I'll have another look at the CSRNG unittests.

tjaychen commented on sw/device/lib/testing/entropy_testutils.c:
```diff
@@ -50,11 +50,11 @@ void entropy_testutils_auto_mode_init(void) {
           {
               .len = 1,
-              .data = {0x00000002},  // Reseed from entropy source only.
+              .data = {0x00000902},  // Reseed from entropy source only.
```
i would probably recommend ORing in `kMubiFalse` into the right bit position, that way in the future if mubi values change it will be slightly more resistant.

vogelpi replied:
This is a good idea. I've changed it in both the testutils and the unittests.

vogelpi commented:
Yep.

vogelpi commented:
Now it all unittests should work.

tjaychen reviewed (approved):
lgtm with one comment.

vogelpi commented:
Thanks @tjaychen for approving this. I've now implemented your change and also removed the hardware change. As @mwbranstad clarified, this wasn't needed. I re-ran the OTBN randomness test locally. Feel free to merge this once CI passes. The nightly regressions can then pick this up.

tjaychen commented:
thanks for fixing this @vogelpi
